src/Map/MapRasterizer_P.cpp

```diff
@@ -441,43 +441,25 @@ void OsmAnd::MapRasterizer_P::rasterizePolygon(
         canvas.drawPath(path, paint);
 }
 
-void OsmAnd::MapRasterizer_P::rasterizePolyline(
+bool OsmAnd::MapRasterizer_P::calcPathByTrajectory(
     const Context& context,
-    SkCanvas& canvas,
-    const std::shared_ptr<const MapPrimitiviser::Primitive>& primitive,
-    bool drawOnlyShadow)
+    const QVector<PointI>& points31,
+    SkPath& path,
+    float offset = 0.0f) const
 {
-    const auto& points31 = primitive->sourceObject->points31;
-    const auto& area31 = context.area31;
-    const auto& env = context.env;
-
-    assert(points31.size() >= 2);
-
-    SkPaint paint = _defaultPaint;
-    if (!updatePaint(context, paint, primitive->evaluationResult, PaintValuesSet::Layer_1, false))
-        return;
-
-    bool ok;
+    int dir = offset < 0 ? -1: 1;
+    offset = abs(offset);
+    bool shift = offset > std::numeric_limits<float>::epsilon() ? true : false;
```

Comment:
this doesn't make any sense "offset" is specified by user, it should be parsed /stored as "0", so here should be no such checks. Checks should be done during parsing if necessary

Reply:
Discussed.
Will be removed.

Reply:
Updated according to discussion.